### DISCUSSION TITLE
Limit the return value of Str::limit to the given limit rather than n + strlen($end)

### DISCUSSION THREAD

**Discussion by MeesterDev** [laravel/framework#38535] [Ideas]
`Str::limit('foobar', 3, '...');` results in "foo...". This is the same length as the the original string, and still 3 over the given limit.

This way the function is not usable for limiting the length of a string that I want to store in a database. One might argue that you can use a lower limit, but I'd be limiting strings that do actually fit, as I've given a limit that's lower than the real limit.

My suggestion is to make `Str::limit('foobar', 5, '...')` return "fo..." instead of "fooba...". I do have doubts about how to handle cases where the limit is lower than `mb_strlen($end)`, but I guess `Str:limit('foobar', 2, '...')` should return "..".

I'd gladly make a PR for this. However, considering the edge case I mentioned and that, if I am nitpicking, it's a breaking change, I thought it best to suggest it as an idea first.

**Comment by wb-joona** [comment]
I also agree. I think that current behavior is counterintuitive, especially since the documentation states: "_The Str::limit method truncates the given string to the specified length_".

IIRC the current documentation should also reflect the fact that `Str::limit` seems to eat away excess whitespace, which makes the output length vary even further:

```
>>> Str::limit('a     ', 6)
=> "a     "
>>> Str::limit('a      ', 6)
=> "a..."
>>> Str::limit('aaaaaa', 6)
=> "aaaaaa"
>>> Str::limit('aaaaaaa', 6)
=> "aaaaaa..."
```
and
```
>>> Str::length(Str::limit('a     ', 6))
=> 6
>>> Str::length(Str::limit('a      ', 6))
=> 4
>>> Str::length(Str::limit('aaaaaa', 6))
=> 6
>>> Str::length(Str::limit('aaaaaaa', 6))
=> 9
```
